cmd/publicize/server.go

```diff
@@ -0,0 +1,237 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	prowconfig "k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+var publicizeRe = regexp.MustCompile(`(?mi)^/publicize\s*$`)
+
+func helpProvider(_ []prowconfig.OrgRepo) (*pluginhelp.PluginHelp, error) {
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: `The publicize plugin is used for merging a pull request into the bonded repository.`,
+	}
+	pluginHelp.AddCommand(pluginhelp.Command{
+		Usage:       "/publicize",
+		Description: "Merge the specific pull request into the bonded repository",
+		WhoCanUse:   "Members of the trusted organization for the repo.",
+		Examples:    []string{"/publicize"},
+	})
+	return pluginHelp, nil
+}
+
+type server struct {
+	webhookTokenGenerator func() []byte
+	githubTokenGenerator  func() []byte
+
+	bonds   map[string]string
+	confirm bool
+
+	gitName  string
+	gitEmail string
+
+	ghc github.Client
+	log *logrus.Entry
+}
+
+func (s *server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	eventType, eventGUID, payload, ok, _ := github.ValidateWebhook(w, r, s.webhookTokenGenerator)
+	if !ok {
+		return
+	}
+	fmt.Fprint(w, "Event received. Have a nice day.")
+
+	if err := s.handleEvent(eventType, eventGUID, payload); err != nil {
+		logrus.WithError(err).Error("Error parsing event.")
+	}
+}
+
+func (s *server) handleEvent(eventType, eventGUID string, payload []byte) error {
+	l := logrus.WithFields(logrus.Fields{"event-type": eventType, github.EventGUID: eventGUID})
+
+	switch eventType {
+	case "issue_comment":
+		var ic github.IssueCommentEvent
+		if err := json.Unmarshal(payload, &ic); err != nil {
+			return err
+		}
+		go func() {
+			if err := s.handleIssueComment(l, ic); err != nil {
+				s.log.WithError(err).WithFields(l.Data).Info("Failed to publicize pull request.")
+			}
+		}()
+	default:
+		logrus.Debugf("skipping event of type %q", eventType)
+	}
+	return nil
+}
+
+func (s *server) handleIssueComment(l *logrus.Entry, ic github.IssueCommentEvent) error {
+	if !ic.Issue.IsPullRequest() || ic.Action != github.IssueCommentActionCreated {
```

Comment:
This should realistically be a generic comment handler and handle all the other cases -- people will want to be able to use a review, etc, like they can with other commands for Prow